x-pack/plugins/observability_solution/investigate_app/public/pages/details/components/add_investigation_item/add_investigation_item.tsx

```diff
@@ -113,10 +108,10 @@ export function AddInvestigationItem({ onItemAdd: onItemAdd, timeRange }: Props)
               ) : (
                 <EsqlWidgetPreview
                   esqlQuery={submittedQuery.esql}
-                  timeRange={timeRange}
-                  onItemAdd={(item) => {
+                  timeRange={globalParams.timeRange}
+                  onItemAdd={async (item) => {
```

Comment:
Instead of passing down to the child component `onItemAdd` and `timeRange` could you get this information from the context directly in the `EsqlWidgetPreview`component?

Reply:
Regarding `onItemAdd` I see you need to call resetState here. But for the timeRange, you could get this info from the context in the EsqlWidgetPreview component.